cpp/src/arrow/dataset/scanner.h

```diff
@@ -136,6 +137,14 @@ struct ARROW_DS_EXPORT ScanOptions {
   /// Parameters which control when the plan should pause for a slow consumer
   acero::BackpressureOptions backpressure =
       acero::BackpressureOptions::DefaultBackpressure();
+
+  /// Parameters which control where the scan starts
```

Comment:
nit: maybe include period at the end of the sentence. Other code comments follow that style.